sdk/core/System.Net.ClientModel/src/Options/RequestOptions.cs

```diff
@@ -11,49 +11,52 @@ namespace System.Net.ClientModel;
 /// the message being sent down the pipeline.  For the duration of pipeline.Send,
 /// this may change some behaviors in various pipeline policies and the transport.
 /// </summary>
-// TODO: Make options freezable
-// Note: I was calling this RequestOptions, but I'm changing it back to RequestOptions.
-public class RequestOptions : PipelineOptions
+public class RequestOptions
 {
-    public virtual void Apply(ClientMessage message)
+    private MessagePipeline? _requestPipeline;
+
+    public RequestOptions() : this(new PipelineOptions())
+    {
+    }
+
+    public RequestOptions(PipelineOptions pipelineOptions)
+    {
+        PipelineOptions = pipelineOptions;
```

Comment:
This should copy the values over instead of storing a reference to the passed-in options